api/environment/index.ts

```diff
@@ -18,6 +18,9 @@ function environmentRoute(req, res) {
     req.session.env = true
     return req.session.save(() => {
       logger.info('new session saved! ', req.session.id)
+      const csrfToken = req.csrfToken()
+      logger.info('token for new session: ', csrfToken)
+      res.cookie('XSRF-TOKEN', csrfToken)
```

Comment:
I would do this in server.ts instead on the 'render' call (as we only need to generate the token once - this api endpoint can get called multiple times). See https://www.npmjs.com/package/csurf#single-page-application-spa

Reply:
Ah good shout!

Reply:
As discussed: Leaving this as is, as the config is always called first, and not in all instances is the index page served through express